rs/types/base_types/src/canister_id.rs

```diff
@@ -121,9 +167,9 @@ impl TryFrom<PrincipalId> for CanisterId {
 impl TryFrom<&[u8]> for CanisterId {
     type Error = CanisterIdError;
     fn try_from(bytes: &[u8]) -> Result<Self, Self::Error> {
-        Self::try_from(
+        Ok(Self::unchecked_from_principal(
             PrincipalId::try_from(bytes).map_err(CanisterIdError::PrincipalIdParseError)?,
-        )
+        ))
```

Comment:
Is this change necessary to be in this PR? I'd expect that we actually care about this once we try the migration of `try_from` to use `try_from_principal_id` internally?

Reply:
Removed.

No, we do not need these changes in this PR (because this is a behavior preserving change).

What I was envisioning is that we'd get to the point where we can comment out `impl CanisterId from PrincipalId { ... }`, and everything would _still_ build (and all tests would still pass). Then, we can be sure that nobody is counting on the old behavior, since nobody is even calling it. Once we get to that point, then, all callers of try_from_principal_id could be converted (back) to CanisterId::try_from.

This is just an example of what you would have to do to reach "nobody is relying on the old behavior".

Yes, the goal state is for other try_from implementations to reduce to the one for PrincipalId.

Ideally, you would be able to take _any_ type that converts to PrincipalId, and be able to (directly) convert it to CanisterId. There might be a way to do this, but [when I tried], I couldn't get it to work.

[when I tried]: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=e023aa043771fdf1beb7af593845a383